### PR TITLE
[Run2_2017] Prepare for V16 Production

### PR DESCRIPTION
**Pull request by aperloff**
Lots of changes needed to run the V16 production:
  1) File lists of added samples since V15
    - Some samples moved from PRODUCTION to VALID status
    - Some more sampels were requested by users
  2) New neff dictionary including the additional samples. A second, offline dictionary of just the added samples is being kept so we don't have to run all of the jobs again.
  3) Updates to some of the scripts used to make the file lists (get_py, get_mcm, makeDicts)
  4) A new tm_common.py file for keeping functions useful to multiple python scripts.

### CODE REVIEW

File: Production/python/get_mcm.py [hunk 1]
```diff
@@ -23,6 +24,46 @@ class col:
     bold = '\033[1m'
     uline = '\033[4m'
 
+class getpyline:
```
Comment (kpedro88) [inline Production/python/get_mcm.py]: a bit odd to have a class called "getpyline", I might just call it "pyline"

File: Production/python/get_mcm.py [hunk 2]
```diff
@@ -47,6 +88,9 @@ def getA(query,debug):
         print jstr
         return None
 
+def parser_callback(option, opt, value, parser):
```
Comment (kpedro88) [inline Production/python/get_mcm.py]: You can get this from: https://github.com/kpedro88/CondorProduction/blob/master/python/parseConfig.py#L5

File: Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
         ['RunIIFall17MiniAODv2.QCD_Pt_3200toInf_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-15to20_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-20to30_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-30to50_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-50to80_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-80to120_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-120to170_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-170to300_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-300to470_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
+        ['RunIIFall17MiniAODv2.QCD_Pt-470to600_MuEnrichedPt5_TuneCP5_13TeV_pythia8'],
```
Comment (kpedro88) [inline Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py:31]: are there higher-pT samples for MuEnriched?

File: Production/test/condorSub/makeDicts.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import os, glob, fnmatch
 from optparse import OptionParser
 
+def parser_callback(option, opt, value, parser):
```
Comment (kpedro88) [inline Production/test/condorSub/makeDicts.py]: see above comment

File: Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py (hunk 3 above)
Comment (aperloff) [reply Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py:31]: Yes, but some aren't complete and some aren't even being produced yet.

File: Production/test/condorSub/makeDicts.py (hunk 4 above)
Comment (aperloff) [reply Production/test/condorSub/makeDicts.py]: Done

File: Production/python/get_mcm.py (hunk 2 above)
Comment (aperloff) [reply Production/python/get_mcm.py]: Done

File: Production/python/get_mcm.py (hunk 1 above)
Comment (aperloff) [reply Production/python/get_mcm.py]: Done

File: Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py (hunk 3 above)
Comment (kpedro88) [reply Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py:31]: What about cross sections?

Comment (aperloff) [reply Production/test/condorSub/dict_RunIIFall17MiniAODv2_qcd_pt.py:31]: They are there now.

**Pull request merged**